openbb_platform/providers/estat/openbb_estat/models/search.py

```diff
@@ -0,0 +1,202 @@
+        ge=1,
+    )
+    updated_date: Optional[str] = Field(
+        default=None,
+        description="Filter by update date in YYYY-MM-DD format.",
+    )
+
+
+class EstatSearchData(SearchData):
+    """e-Stat Search Data."""
+
+    @field_validator("survey_date", "open_date", mode="before", check_fields=False)
+    @classmethod
+    def validate_dates(cls, v):
+        """Validate and parse dates."""
+        if not v:
+            return None
+        try:
+            # e-Stat returns dates as YYYY-MM-DD strings
+            return datetime.strptime(v, "%Y-%m-%d").date()
+        except (ValueError, TypeError):
+            return None
+
+
+class EstatSearchFetcher(Fetcher[EstatSearchQueryParams, List[EstatSearchData]]):
+    """e-Stat Search Fetcher."""
+
+    @staticmethod
+    def transform_query(params: Dict[str, Any]) -> EstatSearchQueryParams:
+        """Transform query parameters."""
+        return EstatSearchQueryParams(**params)
+
+    @staticmethod
+    async def aextract_data(
+        query: EstatSearchQueryParams,
+        credentials: Optional[Dict[str, str]],
+        **kwargs: Any,
+    ) -> List[Dict]:
+        """Extract data from e-Stat getStatsList API."""
+        # pylint: disable=import-outside-toplevel
+        import aiohttp
+
+        api_key = credentials.get("api_key") if credentials else ""
+
+        # Build API parameters
+        params = {
+            "appId": api_key,
+            "lang": "E",  # Request English where available
+            "limit": query.limit,
+        }
+
+        if query.query:
+            params["searchWord"] = query.query
+
+        if query.start_position:
+            params["startPosition"] = query.start_position
+
+        if query.updated_date:
+            # Convert YYYY-MM-DD to YYYY-MM-DD format (e-Stat accepts this)
+            params["updatedDate"] = query.updated_date
+
+        url = "https://api.e-stat.go.jp/rest/3.0/app/json/getStatsList"
+
+        async with aiohttp.ClientSession() as session:
+            async with session.get(url, params=params) as response:
+                if response.status != 200:
+                    raise Exception(
+                        f"e-Stat API request failed with status {response.status}"
+                    )
+
+                data = await response.json()
+
+                if "GET_STATS_LIST" not in data:
+                    raise EmptyDataError("No data returned from e-Stat API")
+
+                stats_list = data["GET_STATS_LIST"]
+
+                # Check for API errors
+                if "RESULT" in stats_list:
+                    result = stats_list["RESULT"]
+                    status = result.get("STATUS", -1)
+                    if status != 0:
+                        error_msg = result.get("ERROR_MSG", "Unknown error")
+                        formatted_error = handle_estat_error(status, error_msg)
+                        raise EmptyDataError(formatted_error)
```

Comment:
We can use this error type, `from openbb_core.provider.utils.errors import UnauthorizedError`, for API key errors, and this will send the correct HTTP status code to the API output.

```json
{"GET_STATS_LIST":{"RESULT":{"STATUS":100,"ERROR_MSG":"Authentication failed. Please check that your appID are correct.","DATE":"2025-11-06T08:36:34.211+09:00"}
```


```suggestion
                        if "Authentication failed." in error_msg:
                            raise UnauthorizedError(formatted_error)
                        raise OpenBBError(f"eStat Error -> {formatted_error}")
```